lib/install/management/finder.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package management
+
+import (
+	"fmt"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/lib/install/validate"
+	"github.com/vmware/vic/lib/metadata"
+	"github.com/vmware/vic/pkg/errors"
+	"github.com/vmware/vic/pkg/vsphere/compute"
+	"github.com/vmware/vic/pkg/vsphere/extraconfig"
+	"github.com/vmware/vic/pkg/vsphere/vm"
+)
+
+func (d *Dispatcher) NewVCHFromID(id string) (*vm.VirtualMachine, error) {
+	var err error
+	var vmm *vm.VirtualMachine
+
+	moref := new(types.ManagedObjectReference)
+	if ok := moref.FromString(id); !ok {
+		message := "Failed to get appliance VM mob reference"
+		log.Errorf(message)
+		return nil, errors.New(message)
+	}
+	ref, err := d.session.Finder.ObjectReference(d.ctx, *moref)
+	if err != nil {
+		if _, ok := err.(*find.NotFoundError); !ok {
+			err = errors.Errorf("Failed to query appliance (%s): %s", moref, err)
+			return nil, err
+		}
+		log.Debugf("Appliance is not found")
```

Comment:
Given this is debug, let's log the ID supplied as well. Can do this with an opening trace call at the top of the function (which we should have for almost all functions):
```
defer trace.End(trace.Begin(id))
```